api/v1alpha1/frontendenvironment_types.go

```diff
@@ -123,6 +123,12 @@ type FrontendEnvironmentSpec struct {
 	// OverwriteCaddyConfig determines if the operator should overwrite
 	// frontend container Caddyfiles with a common core Caddyfile
 	OverwriteCaddyConfig bool `json:"overwriteCaddyConfig,omitempty"`
+	// Enable Push Cache Container
+	EnablePushCache bool `json:"enablePushCache,omitempty"`
+	// S3 Push Cache Bucket
+	PushCacheBucket string `json:"pushCacheBucket,omitempty"`
+	// List of frontends where the push cache job is allowed to run
+	PushCacheAllowlist []string `json:"pushCacheAllowlist,omitempty"`
```

Comment:
I'd rather offload this to the frontend resource, not the environment. Ideally, apps would mark themselves as ready so we do not have to keep track of which are enabled, but have an exact flag that will say if we should try and trigger the job